framework/src/main/java/org/checkerframework/common/basetype/BaseTypeValidator.java

```diff
@@ -86,16 +86,19 @@ public boolean isValid(AnnotatedTypeMirror type, Tree tree) {
     }
 
     /**
-     * Should the top-level declared type be checked?
+     * Should the top-level declared or primitive type be checked?
+     *
+     * <p>If {@code type} is not a declared or primitive type, then this method returns true.
      *
      * <p>Top-level type is not checked if tree is a local variable or an expression tree.
      *
      * @param type AnnotatedTypeMirror being validated
      * @param tree a Tree whose type is {@code type}
-     * @return whether or not the top-level type should be checked
+     * @return whether or not the top-level type should be checked, if {@code type} is a declared or
+     *     primitive type.
      */
     protected boolean shouldCheckTopLevelDeclaredType(AnnotatedTypeMirror type, Tree tree) {
```

Comment:
Could you deprecate this method and introduce a better name?
It makes no sense to use `DeclaredType` in the method name if it actually is also for primitive types.

Reply:
We can't deprecate methods the might be overridden by subclass.  I can rename it and mention it in the change log.

Reply:
You can deprecate it and have the new method call the deprecated method (suppressing the warning).
All CF overrides can change to override the new method and external overrides would still be considered by the new method in this class.
But as this will be a minor release anyways, directly renaming sounds good to me.